include/zephyr/drivers/adc.h

```diff
@@ -231,6 +231,8 @@ IF_ENABLED(CONFIG_ADC_CONFIGURABLE_INPUTS, \
 	(.differential    = DT_NODE_HAS_PROP(node_id, zephyr_input_negative), \
 	 .input_positive  = DT_PROP_OR(node_id, zephyr_input_positive, 0), \
 	 .input_negative  = DT_PROP_OR(node_id, zephyr_input_negative, 0),)) \
+IF_ENABLED(DT_PROP(node_id, zephyr_differential), \
+	(.differential    = true,)) \
```

Comment:
Won't this result in multiple `differential` entries if `CONFIG_ADC_CONFIGURABLE_INPUTS` is enabled?

Reply:
It will. That's why the `.differential = true,` entry is put as the last one, to override the value derived from `zephyr,input-negative`.